contracts/drafts/dao/DAO.sol

```diff
@@ -0,0 +1,117 @@
+    mapping(address => address[]) public backersForIdea;
+    event Here();
+
+    constructor()
+    ERC20Mintable()
+    ERC20MultiDividendable()
+    IssuanceEth(address(this))
+    public
+    {
+        _createState("NEVER");
+        _createTransition("LIVE", "SETUP");
+        _createTransition("FAILED", "SETUP");
+        addMinter(address(this));
+    }
+
+    function transferFunds(address payable _wallet) public onlyOwner {
+        require(currentState == "NEVER", "You can never do this.");
+    }
+
+    function restartFundingRound() public onlyOwner {
+        require(
+            currentState == "LIVE" || currentState == "FAILED",
+            "Initial funding round not ended."
+        );
+        _transition("SETUP");
+    }
+
+    function begMoneyForIdea(uint256 amount, address payable idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            totalAmounts + amount <= address(this).balance,
+            "You beg too much."
+        );
+        begAmount[idea] = amount;
+    }
+
+    function voteForIdea(uint256 votes, address idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            this.balanceOf(msg.sender) - totalVotesByHolder[msg.sender] >= votes,
+            "Not enough power."
+        );
+        totalVotesForIdea[idea] += votes;
+        totalVotesByHolder[msg.sender] += votes;
+        votesForIdeaByHolder[idea][msg.sender] += votes;
+        resolveBackerForIdea(idea, msg.sender);
+    }
+
+    function fundIdea(address payable idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            totalVotesForIdea[idea] >= this.totalSupply().div(2) + 1,
+            "Not enough expressed votes."
+        );
+        // solium-disable-next-line security/no-call-value
+        IssuanceEth(idea).invest.value(begAmount[idea])();
+        totalAmounts -= begAmount[idea];
+        for (uint256 i = 0; i < backersForIdea[idea].length; i++) {
+            totalVotesByHolder[backersForIdea[
+                    idea
+                ][i]] -= votesForIdeaByHolder[
+                    idea
+                ][backersForIdea[idea][i]];
+        }
+        delete totalVotesForIdea[idea];
+        delete begAmount[idea];
+    }
+
+    function getTokensForFundedIdea(address idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        IssuanceEth issuance = IssuanceEth(idea);
+        ERC20Mintable issuanceToken = ERC20Mintable(
+            address(issuance.issuanceToken())
+        );
+        issuance.withdraw();
+    }
+
+    function getReturnsFromTokensOfFundedIdea(address idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        IssuanceEth issuance = IssuanceEth(idea);
+        address dividendToken = address(issuance.issuanceToken());
```

Comment:
I assume that if we just get Ether back this would be a bit simpler.

Reply:
It would make no sense. The DAO model is analogous to a VC fund. The fund invests in a company money (ether) and receives back shares (tokens). Why would it invest ether to receive back, well, (more) ether? It would be amazing to have a contract that when sending ether to it gives you back more.